src/Voron/Impl/LowLevelTransaction.cs

```diff
@@ -806,6 +815,8 @@ public void Dispose()
                 
                 PagerTransactionState.InvokeDispose(_env, ref DataPagerState, ref PagerTransactionState);
                 OnDispose?.Invoke(this);
+
+                _envRecord = null;
```

Comment:
This will prevent from having a cycle (see https://issues.hibernatingrhinos.com/issue/RavenDB-22972/Rising-managed-memory-usage-and-GC-impact-on-7.0-cluster#focus=Comments-67-1056537.0-0)

Reply:
Shouldn't we add a comment explaining how important that is?

Reply:
Yes, we must have it. Fixed